compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -665,6 +694,13 @@ impl std::fmt::Display for Type {
                     write!(f, "{}", binding.borrow())
                 }
             }
+            Type::TypeVariable(binding, TypeVariableKind::Numeric(typ)) => {
+                if let TypeBinding::Unbound(_) = &*binding.borrow() {
+                    write!(f, "{}", typ)
+                } else {
+                    write!(f, "{}", binding.borrow())
+                }
+            }
```

Comment:
Hmm so if we try to display an array of a certain size we'd get `[Field; 4]` but if the size was unknown we'd display `[Field; u32]`.

Seems a bit confusing so maybe we should just stick to the previous formatting which uses the underscore notation other type variables use `[Field; _]`